qiskit_experiments/calibration_management/base_calibrations.py

```diff
@@ -0,0 +1,114 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Base calibrations class."""
+
+from abc import ABC, abstractmethod
+from typing import Dict, Tuple, Union
+
+from qiskit.circuit import Parameter
+from qiskit.pulse import ScheduleBlock
+
+from qiskit_experiments.calibration_management.parameter_value import ParameterValue
+from qiskit_experiments.calibration_management.calibration_key_types import (
+    ParameterKey,
+    ParameterValueType,
+)
+
+
+class BaseCalibrations(ABC):
+    """An abstract base calibration class that defines the methods needed by cal. experiments.
+
+    A calibration experiment uses an instance of `BaseCalibrations` that defines where
+    to get parameter values from and where to save them. This class also defines a method from
+    which to retrieve pulse-schedules.
+    """
+
+    @abstractmethod
+    def add_parameter_value(
+        self,
+        value: Union[int, float, complex, ParameterValue],
+        param: Union[Parameter, str],
+        qubits: Union[int, Tuple[int, ...]] = None,
+        schedule: Union[ScheduleBlock, str] = None,
+    ):
+        """Add a parameter value to the stored parameters.
+
+        This parameter value may be applied to several channels, for instance, all
+        DRAG pulses may have the same standard deviation.
+
+        Args:
+            value: The value of the parameter to add. If an int, float, or complex is given
+                then the timestamp of the parameter value will automatically be generated
+                and set to the current local time of the user.
+            param: The parameter or its name for which to add the measured value.
+            qubits: The qubits to which this parameter applies.
+            schedule: The schedule or its name for which to add the measured parameter value.
+        """
+
+    @abstractmethod
+    def get_parameter_value(
+        self,
+        param: Union[Parameter, str],
+        qubits: Union[int, Tuple[int, ...]],
+        schedule: Union[ScheduleBlock, str, None] = None,
+    ) -> Union[int, float, complex]:
+        """Retrieves the value of a parameter.
+
+        Parameters may be linked. :meth:`get_parameter_value` does the following steps:
```

Comment:
Note that this sentence is not finished. Also, what do you mean with "Parameters may be linked"? There's only one parameter here,

Reply:
This indicates parameter update can affect other schedules since it's linked.